clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -357,6 +361,51 @@ public boolean conditionMet() {
         }
     }
 
+    @Test
+    public void testDescribeConfigs() throws Exception {
+        try (MockKafkaAdminClientEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareMetadataUpdate(env.cluster(), Collections.<String>emptySet());
+            env.kafkaClient().setNode(env.cluster().controller());
+            env.kafkaClient().prepareResponse(new DescribeConfigsResponse(0,
+                Collections.singletonMap(new org.apache.kafka.common.requests.Resource(BROKER, "0"),
+                    new DescribeConfigsResponse.Config(ApiError.NONE,
+                        Collections.<DescribeConfigsResponse.ConfigEntry>emptySet()))));
+            DescribeConfigsResult result2 = env.adminClient().describeConfigs(Collections.singleton(
+                new ConfigResource(ConfigResource.Type.BROKER, "0")));
+            result2.all().get();
+        }
+    }
+
+    @Test
+    public void testDescribeConfigsWithError() throws Exception {
```

Comment:
A couple of things:
1. This test doesn't test the bug that was fixed. The bug only manifests itself if there's a timeout, the broker is not available or something like that.
2. This test hangs occasionally. When it happens, something like the following appears in the logs:

```text
[2017-09-02 08:46:00,129] ERROR [AdminClient clientId=adminclient-23] Internal server error on 0: ignoring call Call(callName=describeBrokerConfigs, deadlineMs=1504338361127) in correlationIdToCall that did not exist in callsInFlight (org.apache.kafka.clients.admin.KafkaAdminClient:292)
[2017-09-02 08:46:00,130] ERROR [AdminClient clientId=adminclient-23] Internal server error on 1: server returned information about unknown correlation ID 0, requestHeader = RequestHeader(apiKey=DESCRIBE_CONFIGS, apiVersion=0, clientId=mockClientId, correlationId=0) (org.apache.kafka.clients.admin.KafkaAdminClient:297)
```

I merged without the test to make the 0.11.0.1 cut, but we should do a follow-up PR to add it.